conn.go

```diff
@@ -410,9 +408,7 @@ func (c *Conn) recv() error {
 		return err
 	}
 
-	if head.stream > c.numStreams {
```

Comment:
The new code only guarantees that there will be `c.numStreams` active streams, but doesn't guarantee what range they will be. They will likely be contiguous, but could be 500-600 if numStreams is 100.